internal/util/vault.go

```diff
@@ -74,113 +81,216 @@ type VaultKMS struct {
 	// vaultPassphrasePath (VPP) used to be added before the "key" of the
 	// secret (like /v1/secret/data/<VPP>/key)
 	vaultPassphrasePath string
+}
+
+// setConfigString fetches a value from a configuration map and converts it to
+// a string.
+//
+// If the value is not available, *option is not adjusted and
+// errConfigOptionMissing is returned.
+// In case the value is available, but can not be converted to a string,
+// errConfigOptionInvalid is returned.
+func setConfigString(option *string, config map[string]interface{}, key string) error {
+	value, ok := config[key]
+	if !ok {
+		return fmt.Errorf("%w: %s", errConfigOptionMissing, key)
+	}
 
-	secrets loss.Secrets
+	s, ok := value.(string)
+	if !ok {
+		return fmt.Errorf("%w: expected string for %q, but got %T",
+			errConfigOptionInvalid, key, value)
+	}
+
+	*option = s
+	return nil
 }
 
-func (vc *vaultConnection) initConnection(kmsID string, config, secrets map[string]string) error {
+// initConnection sets VAULT_* environment variables in the vc.vaultConfig map,
+// these settings will be used when connecting to the Vault service with
+// vc.connectVault().
+//
+// nolint:gocyclo // iterating through many config options, not complex at all.
+func (vc *vaultConnection) initConnection(kmsID string, config map[string]interface{}, secrets map[string]string) error {
 	vaultConfig := make(map[string]interface{})
 	keyContext := make(map[string]string)
 
 	vc.EncryptionKMSID = kmsID
 
-	vaultAddress, ok := config["vaultAddress"]
-	if !ok || vaultAddress == "" {
-		return errors.New("missing 'vaultAddress' for Vault connection")
+	firstInit := (vc.vaultConfig == nil)
+
+	vaultAddress := "" // required
+	err := setConfigString(&vaultAddress, config, "vaultAddress")
+	switch {
+	case errors.Is(err, errConfigOptionInvalid):
+		return err
+	case firstInit && errors.Is(err, errConfigOptionMissing):
+		return err
+	case !errors.Is(err, errConfigOptionMissing):
+		vaultConfig[api.EnvVaultAddress] = vaultAddress
 	}
-	vaultConfig[api.EnvVaultAddress] = vaultAddress
+	// default: !firstInit
 
-	vaultNamespace, ok := config["vaultNamespace"]
-	if !ok || vaultNamespace == "" {
-		vaultNamespace = vaultDefaultNamespace
+	vaultBackendPath := "" // optional
+	err = setConfigString(&vaultBackendPath, config, "vaultBackendPath")
+	if errors.Is(err, errConfigOptionInvalid) {
+		return err
+	}
+	// set the option if the value was not invalid
+	if !errors.Is(err, errConfigOptionMissing) {
+		vaultConfig[vault.VaultBackendPathKey] = vaultBackendPath
+	}
+
+	vaultTLSServerName := "" // optional
+	err = setConfigString(&vaultTLSServerName, config, "vaultTLSServerName")
+	if errors.Is(err, errConfigOptionInvalid) {
+		return err
+	}
+	// set the option if the value was not invalid
+	if !errors.Is(err, errConfigOptionMissing) {
+		vaultConfig[api.EnvVaultTLSServerName] = vaultTLSServerName
+	}
```

Comment:
can we pass the vaultConfig and key to setConfigString so that we don't need to do multiple checks here?
Maybe as a follow-up later also?

Reply:
maybe this can be done with an additional helper, but I do not immediately see how it will be come (much) simpler or cleaner